Explorer/Assets/DCL/VoiceChat/VoiceChatLivekitRoomHandler.cs

```diff
@@ -0,0 +1,347 @@
+            {
+                WeakReference<IAudioStream> stream = voiceChatRoom.AudioStreams.ActiveStream(participant.Identity, publication.Sid);
+
+                if (stream != null) { combinedStreamsAudioSource.RemoveStream(stream); }
+            }
+        }
+
+        private void CloseMedia()
+        {
+            if (!PlayerLoopHelper.IsMainThread) { CloseMediaAsync().Forget(); }
+
+            CloseMediaInternal();
+        }
+
+        private async UniTaskVoid CloseMediaAsync()
+        {
+            if (!PlayerLoopHelper.IsMainThread) { await UniTask.SwitchToMainThread(); }
+
+            CloseMediaInternal();
+        }
+
+        private void CloseMediaInternal()
+        {
+            if (combinedStreamsAudioSource != null)
+            {
+                combinedStreamsAudioSource.Stop();
+                combinedStreamsAudioSource.Reset();
+            }
+
+            monoRtcAudioSource?.Stop();
+            voiceChatRoom.TrackSubscribed -= OnTrackSubscribed;
+            voiceChatRoom.TrackUnsubscribed -= OnTrackUnsubscribed;
+        }
+
+        private void HandleUnexpectedDisconnection()
+        {
+            int remoteCount = voiceChatRoom.Participants.RemoteParticipantIdentities().Count;
+
+            if (remoteCount == 0)
+            {
+                ReportHub.Log(ReportCategory.VOICE_CHAT, "[VoiceChatLivekitRoomHandler] No remote participants in room, skipping reconnection attempts");
+                voiceChatCallStatusService.HandleLivekitConnectionFailed();
+                return;
+            }
+
+            reconnectionAttempts = 0;
+            reconnectionCts = reconnectionCts.SafeRestart();
+            ReportHub.Log(ReportCategory.VOICE_CHAT, "[VoiceChatLivekitRoomHandler] Starting reconnection attempts");
+            AttemptReconnectionAsync(reconnectionCts.Token).Forget();
+        }
+
+        private async UniTaskVoid AttemptReconnectionAsync(CancellationToken ct)
+        {
+            while (!ct.IsCancellationRequested)
+            {
+                if (reconnectionAttempts >= configuration.MaxReconnectionAttempts)
+                {
+                    reconnectionAttempts = 0;
+                    ReportHub.Log(ReportCategory.VOICE_CHAT, $"[VoiceChatLivekitRoomHandler] Max reconnection attempts ({configuration.MaxReconnectionAttempts}) reached - calling HandleLivekitConnectionFailed");
+                    voiceChatCallStatusService.HandleLivekitConnectionFailed();
+                    return;
+                }
+
+                reconnectionAttempts++;
+                ReportHub.Log(ReportCategory.VOICE_CHAT, $"[VoiceChatLivekitRoomHandler] Reconnection attempt {reconnectionAttempts}/{configuration.MaxReconnectionAttempts}");
+                await UniTask.Delay(configuration.ReconnectionDelayMs, cancellationToken: ct);
+
+                bool success = await roomHub.VoiceChatRoom().TrySetConnectionStringAndActivateAsync(voiceChatCallStatusService.RoomUrl);
```

Comment:
Why is it handled on the level of the handler and not on the level of the room? Can you contact me, maybe I don't fully comprehend the flow